jaxrs-api/src/main/java/jakarta/ws/rs/core/Variant.java

```diff
@@ -220,14 +217,13 @@ public boolean equals(final Object obj) {
             return false;
         }
         final Variant other = (Variant) obj;
-        if (this.language != other.language && (this.language == null || !this.language.equals(other.language))) {
+        if (!Objects.equals(this.language, other.language)) {
             return false;
         }
-        if (this.mediaType != other.mediaType && (this.mediaType == null || !this.mediaType.equals(other.mediaType))) {
+        if (!Objects.equals(this.mediaType, other.mediaType)) {
             return false;
         }
-        // noinspection StringEquality
-        return this.encoding == other.encoding || (this.encoding != null && this.encoding.equals(other.encoding));
+        return (!Objects.equals(this.encoding, other.encoding));
```

Comment:
Does the new code return the negation of the original code?

Reply:
Good catch! A typo, indeed.

Reply:
Thanks for your review! I fixed the typo in https://github.com/eclipse-ee4j/jaxrs-api/pull/1075/commits/3c9220686c543dd9d0b8937819b0f9911d130d6c.